### PR TITLE
chore: add ArrowMixin improvements, add Datepicker local/global switch

### PR DESCRIPTION
**Pull request by daKmoR**
## What I did

- ArrowMixin needs to extend styles and not overwrite them
- ArrowMixin add an `_arrowNodeTemplate` which can be used to only override the arrow content
- InputDatepicker switch between bottomsheet style (on mobile) popover (on desktop/table with screensize bigger then 600px)

### CODE REVIEW

File: packages/input-datepicker/src/LionInputDatepicker.js [hunk 1]
```diff
@@ -280,6 +290,18 @@ export class LionInputDatepicker extends ScopedElementsMixin(OverlayMixin(LionIn
     `;
   }
 
+  _setupOverlayCtrl() {
+    super._setupOverlayCtrl();
+
+    this._overlayCtrl.addEventListener('before-show', () => {
+      if (window.innerWidth >= 600) {
+        this._overlayCtrl.updateConfig(this._defineOverlayConfig());
```
Comment (ruphin) [inline packages/input-datepicker/src/LionInputDatepicker.js:298]: I feel like the configuration of the overlay is too spread out across different parts. I would add the `window.innerWidth` switch to `_defineOverlayConfig()`:

```js
_defineOverlayConfig() {
  if (window.innerWidth >= 600) {
    return {
      ...withModalDialogConfig(),
      hidesOnOutsideClick: true,
      ...super._defineOverlayConfig(),
      popperConfig: {
        ...super._defineOverlayConfig().popperConfig,
        placement: 'bottom',
      },
    };
  } else {
    return withBottomSheetConfig();
  }
}
```

and

```js
this._overlayCtrl.addEventListener('before-show', () => {
  this._overlayCtrl.updateConfig(this._defineOverlayConfig());
}
```

Comment (daKmoR) [reply packages/input-datepicker/src/LionInputDatepicker.js]: changed

File: packages/input-datepicker/src/LionInputDatepicker.js [hunk 2]
```diff
@@ -280,17 +290,36 @@ export class LionInputDatepicker extends ScopedElementsMixin(OverlayMixin(LionIn
     `;
   }
 
+  _setupOverlayCtrl() {
+    super._setupOverlayCtrl();
+
+    this.__datepickerBeforeShow = () => {
+      this._overlayCtrl.updateConfig(this._defineOverlayConfig());
```
Comment (tlouisse) [inline packages/input-datepicker/src/LionInputDatepicker.js:297]: Is it needed to always redefine the config?

File: packages/input-datepicker/src/LionInputDatepicker.js [hunk 3]
```diff
@@ -280,17 +290,36 @@ export class LionInputDatepicker extends ScopedElementsMixin(OverlayMixin(LionIn
...
    * @desc overrides default configuration options for this component
    * @returns {Object}
    */
   // eslint-disable-next-line class-methods-use-this
   _defineOverlayConfig() {
-    return {
-      ...withModalDialogConfig(),
-      hidesOnOutsideClick: true,
-    };
+    if (window.innerWidth >= 600) {
```
Comment (tlouisse) [inline packages/input-datepicker/src/LionInputDatepicker.js:309]: Should we define a hardcoded breakpoint value on the lion layer?

File: packages/input-datepicker/src/LionInputDatepicker.js [hunk 4]
```diff
@@ -280,17 +290,36 @@ export class LionInputDatepicker extends ScopedElementsMixin(OverlayMixin(LionIn
...
+      this.hasArrow = true;
+      return {
+        ...withModalDialogConfig(),
+        hidesOnOutsideClick: true,
+        ...super._defineOverlayConfig(),
```
Comment (tlouisse) [inline packages/input-datepicker/src/LionInputDatepicker.js:314]: This adds placementMode 'local'? https://github.com/ing-bank/lion/blob/master/packages/overlays/src/OverlayMixin.js#L94

Maybe we can make it more explicit by adding `placementMode: 'local'` instead?

Comment (daKmoR) [reply packages/input-datepicker/src/LionInputDatepicker.js:314]: this actually adds a lot more from the `ArrowMixin`... for placementMode alone yes copying would be better

File: packages/input-datepicker/src/LionInputDatepicker.js (hunk 2 above)
Comment (daKmoR) [reply packages/input-datepicker/src/LionInputDatepicker.js:297]: I'm not sure if checking a specific overlayCtrl config option would be "better"...

imho setting it every time is good enough for now

File: packages/input-datepicker/src/LionInputDatepicker.js (hunk 3 above)
Comment (daKmoR) [reply packages/input-datepicker/src/LionInputDatepicker.js:309]: probable good enough for now... if someone needs it configurable then a feature request with some discussion will need to happen... (as we currently have no central breakpoints defined anywhere in lion)

and that would definitely be a separate pull request

File: packages/input-datepicker/src/LionInputDatepicker.js (hunk 4 above)
Comment (tlouisse) [reply packages/input-datepicker/src/LionInputDatepicker.js:314]: Ah, indeed 👍

File: packages/input-datepicker/src/LionInputDatepicker.js (hunk 2 above)
Comment (tlouisse) [reply packages/input-datepicker/src/LionInputDatepicker.js:297]: Yeah, we do it once per switch on our extension layer.
Let's make a small improvement issue 👍

**Review by tlouisse — APPROVED** [review]

**Pull request merged**